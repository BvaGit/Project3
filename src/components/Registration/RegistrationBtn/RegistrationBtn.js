import * as React from 'react'

import Button from "../../Button";

const RegistrationBtn = ({ regAuthRequest, isError }) => {
    return (
        <Button
            type="button"
            text="SIGN UP"
            onClick={regAuthRequest}
            disabled={!isError}
        />
    )
};

export default RegistrationBtn;
