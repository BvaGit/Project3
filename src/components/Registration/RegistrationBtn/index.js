import {connect} from "react-redux";

import RegistrationBtn from "./RegistrationBtn";
import {regAuthRequest} from "../../../store/registration/asyncActions";
import {getIsValidRegFields} from "../../../store/registration/selectors";

const mapStateToProps = (state) => ({
    isError: getIsValidRegFields(state),
});

const mapDispatchToProps = (dispatch) => ({
    regAuthRequest: () => dispatch(regAuthRequest()),
});

export default connect(mapStateToProps, mapDispatchToProps)(RegistrationBtn)
