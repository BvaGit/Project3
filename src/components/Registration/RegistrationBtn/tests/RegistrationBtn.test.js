import React from 'react';
import { shallow, mount } from 'enzyme';
import RegistrationBtn from '../RegistrationBtn';


describe('RegistrationBtn', () => {
    it("Should match snapshot", () => {
        const component = shallow(<RegistrationBtn />)
        expect(component.html()).toMatchSnapshot()
    })
})