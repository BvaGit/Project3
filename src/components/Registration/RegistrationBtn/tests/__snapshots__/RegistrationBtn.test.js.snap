// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationBtn Should match snapshot 1`] = `"<button type=\\"button\\" class=\\"button\\" disabled=\\"\\">SIGN UP</button>"`;
